### PR TITLE
Add support for partitioned check

### PR DESCRIPTION
**Pull request by layanto**
First small step in supporting partitioned database feature of couchDB v3+

### CODE REVIEW

**Comment by layanto** [comment]
@redgeoff I have fixed the node test but browser test still broken. This is not relating to the change I have made but rather to the tighter default security of couchDB v3+. I think the reason why the test passed in node but failed in browser is to do with handling of cookie by the browser. In node, instantiating a new slouch instance resets everything and logins as admin user. But in browser, maybe I need to do something else to reset the previous user session?

**Comment by layanto** [comment]
@redgeoff I think ready to merge now. Not sure if I have correctly fixed the broken tests. I added destroySession in two user test spec to clear the session before the next test, else the afterEach for the test failed - I think maybe because failing to destroy username because attempting to destroy username user last session user (i.e. not admin).

File: test/spec/user.js [hunk 1]
```diff
@@ -185,6 +184,9 @@ describe('user', function () {
       session.userCtx.name.should.eql(username);
       session.userCtx.roles.should.eql(['testrole1']);
       (session.cookie === undefined).should.eql(false);
+      user.destroySession().then(function (response) {
```
Comment (redgeoff) [inline test/spec/user.js:187]: I think a “return” is needed here

**Comment by layanto** [comment]
@redgeoff Implemented change as requested

**Pull request merged**